code/game/turfs/closed/walls.dm

```diff
@@ -60,6 +60,16 @@
 /turf/closed/wall/attack_tk()
 	return
 
+/turf/closed/wall/turf_destruction(damage_flag, additional_damage)
+	var/previous_type = type
+	dismantle_wall(prob((additional_damage - 50) / 3), TRUE)
+	// If we scrape away into a turf of the same type, don't go any deeper.
+	if (type == previous_type)
+		return
+	// Cascade turf damage downwards on destruction
+	if (additional_damage > 0)
+		take_damage(additional_damage, BRUTE, damage_flag, FALSE)
```

Comment:
Due to the way turfs work, changing turf actually change src. Because of this turfs will never be deleted, as making a new turf simply updates the current reference. This is how src.type can be different before and after calling dismantle wall.